### PR TITLE
ENC:disable multi slice feature of AVC encoder on APL/KBL/SKL

### LINKED ISSUE #117: Multi Slice feature of AVC encoder is broken on SKL/APL/KBL

### PR DESCRIPTION
**Pull request by QuPengfei**
this is WA to disable multi slcie feature as the improved AVC encoder does not support it.

Fixes #117

Signed-off-by: Pengfei Qu <Pengfei.Qu@intel.com>

we will add back this feature in Q2 release

### CODE REVIEW

File: src/i965_drv_video.c [hunk 1]
```diff
@@ -1078,6 +1078,27 @@ i965_GetConfigAttributes(VADriverContextP ctx,
 
             break;
 
+        case VAConfigAttribEncMaxSlices:
+            if (entrypoint == VAEntrypointEncSlice ||
+                entrypoint == VAEntrypointEncSliceLP) {
```
Comment (xhaihao) [inline src/i965_drv_video.c]: The WA shouldn't disable multiple slice support for  VAEntrypointEncSliceLP

Comment (xhaihao) [inline src/i965_drv_video.c]: The WA shouldn't disable multiple slice for VAEntrypointEncSliceLP

File: src/i965_drv_video.c [hunk 2]
```diff
@@ -1078,6 +1078,27 @@ i965_GetConfigAttributes(VADriverContextP ctx,
...
+                if ((profile == VAProfileH264ConstrainedBaseline ||
+                    profile == VAProfileH264Main ||
+                    profile == VAProfileH264High) ||
+                    profile == VAProfileH264StereoHigh ||
+                    profile == VAProfileH264MultiviewHigh ||
+                    profile == VAProfileHEVCMain ||
+                    profile == VAProfileHEVCMain10) {
```
Comment (xhaihao) [inline src/i965_drv_video.c]: Is multiple slice also disabled for HEVC encoding? if so, please update your commit log

File: src/i965_drv_video.c (hunk 1 above)
Comment (xhaihao) [inline src/i965_drv_video.c]: The WA shouldn't disable multiple slice for VAEntrypointEncSliceLP

File: src/i965_drv_video.c (hunk 2 above)
Comment (xhaihao) [inline src/i965_drv_video.c]: is multi slice disabled for HEVC? if so, please update your commit log or split the patch into 2 small patches

File: src/i965_encoder.c [hunk 3]
```diff
@@ -1460,6 +1467,11 @@ intel_enc_hw_context_init(VADriverContextP ctx,
             if (encoder_context->codec == CODEC_H264)
                 encoder_context->context_roi = 1;
         }
+        if (obj_config->attrib_list[i].type == VAConfigAttribEncMaxSlices) {
```
Comment (xhaihao) [inline src/i965_encoder.c]: VAConfigAttribEncMaxSlices is a read-only attribute, user needn't provide VAConfigAttribEncMaxSlices when creating a VA context. so it is possible that the attrib list doesn't have VAConfigAttribEncMaxSlices.

Comment (scott-d-phillips) [inline src/i965_encoder.c]: Agree with Haihao. This breaks multi-slice encoding on pre-gen9.

**Comment by scott-d-phillips** [comment]
> we will add back this feature in Q2 release

Take the Q2 commitment out of the commit message.

File: src/i965_drv_video.c (hunk 1 above)
Comment (QuPengfei) [inline src/i965_drv_video.c]: As the VAConfigAttribEncMaxSlices is added at the first time, I am not sure the VAEntrypointEncSliceLP support/test multi slices. VAEntrypointEncSliceLP support the multi slice, right?

Comment (yakuizhao) [inline src/i965_drv_video.c]: @QuPengfei , The EncSliceLP is for VDENC. And the multi-slices are supported  although it is not exported.

Comment (QuPengfei) [inline src/i965_drv_video.c]: sure

**Pull request merged**